modules/apps/27-interchain-accounts/controller/keeper/handshake_test.go

```diff
@@ -30,6 +30,24 @@ func (suite *KeeperTestSuite) TestOnChanOpenInit() {
 			},
 			true,
 		},
+		{
+			"success - previous active channel closed",
+			func() {
+				suite.chainA.GetSimApp().ICAControllerKeeper.SetActiveChannelID(suite.chainA.GetContext(), ibctesting.FirstConnectionID, path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID)
+
+				counterparty := channeltypes.NewCounterparty(path.EndpointB.ChannelConfig.PortID, path.EndpointB.ChannelID)
+				channel := channeltypes.Channel{
+					State:          channeltypes.CLOSED,
+					Ordering:       channeltypes.ORDERED,
+					Counterparty:   counterparty,
+					ConnectionHops: []string{path.EndpointA.ConnectionID},
+					Version:        TestVersion,
```

Comment:
Hmm, shouldn't the version set in state on the closed channel also include the populated address field? 

This would mean that you cannot re-open a channel using `RegisterInterchainAccount` tho as the metadata built explicitly with an address: https://github.com/cosmos/ibc-go/blob/main/modules/apps/27-interchain-accounts/controller/keeper/account.go#L45-L53

Reply:
(whoops, didn't see marked as draft)

Reply:
I had opened it as r4r and then decided to change it up